Topic 1/src/point6/TransactionCounter.java

```diff
@@ -0,0 +1,27 @@
+package point6;
+/**
+ * 
+ * @author cris14
+ * Singleton Pattern
+ *
+ */
+public class TransactionCounter {
+	private static TransactionCounter transaction;
+	
+	private int sequence;
+	
+	private TransactionCounter(){
+		this.sequence= 0;
+	}
+
+	public static TransactionCounter getTransaction(){
```

Comment:
[High] You should *syncronized* this method to be threadsafe. You can check this example: http://www.javabeginner.com/learn-java/java-singleton-design-pattern.
Also you can read about *volatile* variables.